spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/config/EnableSpannerRepositories.java

```diff
@@ -120,7 +120,7 @@
 	 *
 	 * @return
 	 */
-	String spannerOperationsRef() default "spannerOperations";
+	String spannerOperationsRef() default "spannerTemplate";
```

Comment:
`spannerTemplateRef` then?

At least that is what I see in the `@EnableMongoRepositories`:
```
	/**
	 * Configures the name of the {@link MongoTemplate} bean to be used with the repositories detected.
	 *
	 * @return
	 */
	String mongoTemplateRef() default "mongoTemplate";
```

Reply:
I'd rather not change this since it requires a lot of cascading changes.

Reply:
???
You just do `Refactoring -> Rename...` and every consumer of this property will be change respectively.

Again: we lose consistency if we stick with the name as it is right now.
Pay attention to the `@EnableRedisRepositories` as well:
```
	/**
	 * Configures the bean name of the {@link RedisOperations} to be used. Defaulted to {@literal redisTemplate}.
	 *
	 * @return
	 */
	String redisTemplateRef() default "redisTemplate";
```

Reply:
I think this makes sense to not change. The significance of this name is in `SpannerRepositoryConfigurationExtension`. "spannerOperationsRef" I believe is how the repository factory knows where to get an instance of the SpannerOperations needed to create a SimpleSpannerRepository. Following the guideline of the consumer (in this case the SimpleSpannerRepository) needing only a SpannerOperations and not the more specific SpannerTemplate, I think it makes sense that the factory use a "spannerOperationsRef" rather than a "spannerTemplateRef".

The value within being "spannerTemplate" just tells which bean function to call to get a SpannerOperations for making the SimpleSpannerRepository.

Reply:
I think this must be renamed, for consistency with all existing Spring Data projects.
One more sample: https://docs.spring.io/spring-data/cassandra/docs/current/api/org/springframework/data/cassandra/repository/config/EnableCassandraRepositories.html#cassandraTemplateRef--

That one indeed can stay as `operations`. Here is a sample:
```
public void postProcess(BeanDefinitionBuilder builder, AnnotationRepositoryConfigurationSource config) {

	AnnotationAttributes attributes = config.getAttributes();

	builder.addPropertyReference("mongoOperations", attributes.getString("mongoTemplateRef"));
	builder.addPropertyValue("createIndexesForQueryMethods", attributes.getBoolean("createIndexesForQueryMethods"));
}
```

@mp911de,

would you mind to share your feedback on the matter, please?

Reply:
That works for me too.